diff --git a/web/core/lib/Drupal/Core/Form/FormHelper.php b/web/core/lib/Drupal/Core/Form/FormHelper.php
index ffced04686..f9bcc36061 100644
--- a/web/core/lib/Drupal/Core/Form/FormHelper.php
+++ b/web/core/lib/Drupal/Core/Form/FormHelper.php
@@ -205,7 +205,7 @@ public static function processStates(array &$elements) {
     // still want to be able to show/hide them. Since there's no actual HTML
     // input element available, setting #attributes does not make sense, but a
     // wrapper is available, so setting #wrapper_attributes makes it work.
-    $key = ($elements['#type'] == 'item') ? '#wrapper_attributes' : '#attributes';
+    $key = isset($elements['#type']) && $elements['#type'] === 'item' ? '#wrapper_attributes' : '#attributes';
     $elements[$key]['data-drupal-states'] = Json::encode($elements['#states']);
   }

